core/spec/lib/spree/core/controller_helpers/auth_spec.rb

```diff
@@ -45,6 +45,25 @@ def controller.index
       expect(response.headers["Set-Cookie"]).to match(/guest_token.*HttpOnly/)
       expect(response.cookies['guest_token']).not_to be_nil
     end
+
+    context 'with guest_token_cookie_options configured' do
+      it 'sends cookie with these options' do
+        stub_spree_preferences(guest_token_cookie_options: {
+          domain: :all,
+          path: '/api'
+        })
+        get :index
+        expect(response.headers["Set-Cookie"]).to match(/domain=\.test\.host; path=\/api/)
+      end
+
+      it 'never overwrites httponly' do
```

Comment:
Just saw this test related to my previous comment. Why do we need this?

Reply:
We do not want that devs shoot theirselves in the foot. Setting `httponly` to `false` means the cookie is readable by javascript in the client browser, making the door open for session highjacking attacks.